api/node/index.ts

```diff
@@ -930,17 +930,22 @@ var globalEventLoop: EventLoop = new EventLoop;
  * If the event loop is already running, then this function returns the same promise as from
  * the earlier invocation.
  *
- * @param runningCallback Optional callback that's invoked once when the event loop is running.
+ * @param args Optional additional arguments.
+ * @param args.runningCallback Optional callback that's invoked once when the event loop is running.
  *                         The function's return value is ignored.
+ * @param args.quitOnLastWindowClosed if set to `true` event loop is quit after last window is closed otherwise
+ *                          it is closed after {@link quitEventLoop} is called.
+ *                          This is useful for system tray applications where the application needs to stay alive even if no windows are visible.
+ *                          (default true).
  *
  * Note that the event loop integration with Node.js is slightly imperfect. Due to conflicting
  * implementation details between Slint's and Node.js' event loop, the two loops are merged
  * by spinning one after the other, at 16 millisecond intervals. This means that when the
  * application is idle, it continues to consume a low amount of CPU cycles, checking if either
  * event loop has any pending events.
  */
-export function runEventLoop(runningCallback?: Function): Promise<unknown> {
-    return globalEventLoop.start(runningCallback)
+export function runEventLoop(args?: { runningCallback?: Function, quitOnLastWindowClosed?: boolean } ): Promise<unknown> {
```

Comment:
You can write this as

`export function runEventLoop(callbackOrOptions: Function | { callback?: Function; runUntilForever?: boolean }): Promise<unknown>`

and then you can do the type narrowing like this:

```typescript
    if (callbackOrOptions instance Function) {
        // now callbackOrOptions is a function callabke
    } else {
        // treat it as the object
    }
```

Reply:
Thank you for the hint. That the same way I also thought about to solve it ;-)

Reply:
changed.